policy_definitions/tagging/policy_set_definition_qby_require_tags.tmpl.json

```
@@ -0,0 +1,39 @@
+{
+    "name": "QBY-Define-Require-Tags",
+    "type": "Microsoft.Authorization/policySetDefinitions",
+    "apiVersion": "2021-06-01",
+    "scope": null,
+    "properties": {
+      "policyType": "Custom",
+      "displayName": "Define require tags to resources",
+      "description": "Require resources and resource groups to have conceptual required tags.",
+      "metadata": {
+        "version": "1.0.0",
+        "category": "Tags"
+      },
+      "parameters": {},
+      "policyDefinitions": ${jsonencode(flatten([for tag_name in tags: [
+        {
+            "policyDefinitionReferenceId": "Require a tag on resource groups",
```

The referenceId needs to be unique. It is used to reference this specific policy assignment in this policy
```suggestion
            "policyDefinitionReferenceId": "${tag_name}-ResourceGroups",
```

---

Changed as suggested.